plugins/experimental/parent_select/strategy.cc

```diff
@@ -71,16 +71,15 @@ PLNextHopSelectionStrategy::Init(const YAML::Node &n)
   bool self_host_used = false;
 
   try {
+    // scheme is optional, and strategies with no scheme will match hosts with no scheme
     if (n["scheme"]) {
       auto scheme_val = n["scheme"].Scalar();
       if (scheme_val == "http") {
         scheme = PL_NH_SCHEME_HTTP;
       } else if (scheme_val == "https") {
         scheme = PL_NH_SCHEME_HTTPS;
       } else {
-        scheme = PL_NH_SCHEME_NONE;
-        PL_NH_Note("Invalid 'scheme' value, '%s', for the strategy named '%s', setting to PL_NH_SCHEME_NONE", scheme_val.c_str(),
-                   strategy_name.c_str());
+        PL_NH_Note("Invalid scheme '%s' for strategy '%s', setting to NONE", scheme_val.c_str(), strategy_name.c_str());
```

Comment:
Since this isn't a constructor, seem slightly fragile to assume data member contain values set at construction.  Hopefully no one ever assumes this function can re-init as well as init.

Reply:
I didn't write the func, it's inherited from core. But it has to exist because the object is constructed by the YAML library.

I think `Init` is effectively like a constructor, because we can't construct, since we can't prevent the YAML library from doing the construction.

But, I think `Init` does make that assumption. There are probably other data it doesn't reset either. I don't object to making Init safe in case someone re-init's an object, and resetting all members to their defaults, but that's probably a larger change and might be better as its own PR